core/listings_test.go

```diff
@@ -12,3 +14,77 @@ func TestFactoryCryptoListingCoinDivisibilityMatchesConst(t *testing.T) {
 		t.Fatal("DefaultCoinDivisibility constant has changed. Please update factory value.")
 	}
 }
+
+func TestValidShippingRegion(t *testing.T) {
+	check := map[int32]bool{
+		// NA
+		0: true,
+		// continents
+		501: true,
+		502: true,
+		503: true,
+		504: true,
+		505: true,
+		506: true,
+		507: true,
+		508: true,
+		// !exist
+		509: true,
+		510: true,
+		511: true,
+		// some random numbers
+		5678:   true,
+		123456: true,
+	}
```

Comment:
Instead of specifying a bool, each example could point at the expected error message it should receive. Valid options can be tested here as well by pointing to a `nil` error. Appreciate the thorough test on both positive and negative values.